### ISSUE TITLE
Parse the "error" field in IndexResponse struct

### ISSUE THREAD

**Issue by aubm** [olivere/elastic#400]
Please use the following questions as a guideline to help me answer
your issue/question without further inquiry. Thank you.

### Which version of Elastic are you using?

[ ] elastic.v5 (for Elasticsearch 5.x)

### Please describe the actual behavior

When digging into the source code, I came accross this "TODO": https://github.com/olivere/elastic/blob/7838d16a32308812791bfd51b43979f725425c6a/index.go#L283

Unfortunately I can't find any way to determine if the index operation succeeded.
Any idea when this will be implemented?

Maybe I can provide a PR, if you will let me know if there are some specifics that I should take in consideration.

For the record, one possible unsuccessful JSON response I use in one of my test cases is the following:

```json
{
  "error": {
    "root_cause": [
      {
        "type": "mapper_parsing_exception",
        "reason": "failed to parse [id]"
      }
    ],
    "type": "mapper_parsing_exception",
    "reason": "failed to parse [id]",
    "caused_by": {
      "type": "json_parse_exception",
      "reason": "..."
    }
  },
  "status": 400
}
```

**Comment by olivere** [comment]
Hi. Thanks for reporting. I'm not sure why you don't get an `error` but a response when Index API returns with status 400. You should get an error, propertly destructed into an [`Error`](https://github.com/olivere/elastic/blob/release-branch.v5/errors.go#L63).

Can you share your code or create a reproducible snippet?

**Comment by aubm** [comment]
Hey, thank you for your quick feedback.

While producing a snippet to demonstrate my issue, I found out that the problem was on me.
My test was not properly set, my mock server did respond with the above JSON, but I forgot to set the HTTP respond code to 400. So the `checkResponse` function will never produce an error.

The "TODO" note in the `IndexResponse` struct made me think that the feature was not ready yet, so I stopped digging.

My bad :)
